### PR TITLE
bugfix/#6074,#6075,#6076,#6077,#6078-Changing-the-access-to-elements-of-admin-sidemenu-according-to-employees-roles

### PR DESCRIPTION
**Pull request by NadiiaStoiko**
#6074,#6075,#6077,#6078 - Changing the access to elements of admin sidemenu according to employees positions and roles
#6076 - Remove opportunity to edit order if employee Driver, Navigator or Logistician

### CODE REVIEW

**Review by KKatiaa — APPROVED** [review]

**Review by mdashko — APPROVED** [review]

File: src/app/shared/ubs-base-sidebar/ubs-base-sidebar.component.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { Router } from '@angular/router';
   styleUrls: ['./ubs-base-sidebar.component.scss']
 })
 export class UbsBaseSidebarComponent implements AfterViewInit, AfterViewChecked, OnDestroy {
-  private destroySub: Subject<boolean> = new Subject<boolean>();
+  public destroySub: Subject<boolean> = new Subject<boolean>();
```
Comment (hnativlyubomyr) [inline src/app/shared/ubs-base-sidebar/ubs-base-sidebar.component.ts:26]: why public?

File: src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts [hunk 2]
```diff
@@ -106,6 +113,45 @@ export class UbsAdminOrderComponent implements OnInit, OnDestroy, AfterContentCh
...
     });
     this.getOrderInfo(this.orderId, false);
+    this.ubsAdminEmployeeService.employeePositions$.pipe(takeUntil(this.destroy$)).subscribe((employeePositions) => {
+      if (employeePositions.length) {
+        this.authoritiesSubscription(employeePositions);
+      }
+    });
+  }
+
+  private authoritiesSubscription(positions) {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts:123]: add positions type here

File: src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts [hunk 3]
```diff
@@ -106,6 +113,45 @@ export class UbsAdminOrderComponent implements OnInit, OnDestroy, AfterContentCh
...
+      employeePositionsName.SuperAdmin,
+      employeePositionsName.Admin,
+      employeePositionsName.CallManager,
+      employeePositionsName.ServiceManager
+    ];
+    let isThisPositionCanEdit = false;
+    let isThisRoleCanEdit = false;
+    if (this.employeePositions) {
+      isThisPositionCanEdit =
+        this.employeePositions.filter((positionsItem: string) => positionsForEditOrder.includes(positionsItem)).length > 0;
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts]: replace length > 0 => length

File: src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts [hunk 4]
```diff
@@ -106,6 +113,45 @@ export class UbsAdminOrderComponent implements OnInit, OnDestroy, AfterContentCh
...
+    }
+
+    if (this.employeeAuthorities) {
+      isThisRoleCanEdit =
+        this.employeeAuthorities.filter((authoritiesItem) => authoritiesItem === abilityEditAuthorities.orders).length > 0;
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts]: the same

File: src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts [hunk 5]
```diff
@@ -1,19 +1,147 @@
...
+  ];
+
+  private employeesCertViewerArr = [
+    employeePositionsName.SuperAdmin,
+    employeePositionsName.Admin,
+    employeePositionsName.CallManager,
+    employeePositionsName.ServiceManager,
+    employeePositionsName.Logistician
+  ];
+
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts:39]: viewersArr = [...employeesCertViewerArr,   employeePositionsName.Navigator]

File: src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts [hunk 6]
```diff
@@ -1,19 +1,147 @@
...
+      if (rights.authorities.length) {
+        this.changeListElementsDependOnPermissions(positions, rights.authorities);
+      }
+    });
+  }
+
+  private positionFilterUtil(posArr: Array<string>): boolean {
+    if (this.employeePositions) {
+      const result = this.employeePositions.filter((positionsItem) => posArr.includes(positionsItem));
+      return result.length > 0;
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts]: return !!result.length

File: src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts [hunk 7]
```diff
@@ -1,19 +1,147 @@
...
+    }
+  }
+
+  private authoritiesFilterUtil(authority: string): boolean {
+    if (this.employeeAuthorities) {
+      const result = this.employeeAuthorities.filter((authoritiesItem) => authoritiesItem === authority);
+      return result.length > 0;
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts]: the same

**Review by hnativlyubomyr — APPROVED** [review]

File: src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts (hunk 2 above)
Comment (NadiiaStoiko) [reply src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts:123]: Thank you. Positions type was added.

File: src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts (hunk 3 above)
Comment (NadiiaStoiko) [reply src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts]: Thank you. Fix was added.

File: src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts (hunk 4 above)
Comment (NadiiaStoiko) [reply src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts]: Thank you. Fix was added.

File: src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts (hunk 5 above)
Comment (NadiiaStoiko) [reply src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts:39]: Thank you. Fixed it.

File: src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts (hunk 6 above)
Comment (NadiiaStoiko) [reply src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts]: Thank you. Changed it.

File: src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts (hunk 7 above)
Comment (NadiiaStoiko) [reply src/app/ubs/ubs-admin/components/ubs-admin-sidebar/ubs-admin-sidebar.component.ts]: Thank you. Changed it.

**Review by katia9261 — APPROVED** [review]

**Review by Tadei-Mykola — APPROVED** [review]

**Review by OksanaFilipchuk — APPROVED** [review]

**Pull request merged**